neo4j-to-neptune/src/main/java/com/amazonaws/services/neptune/util/GzipCompressUtils.java

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+A copy of the License is located at
+    http://www.apache.org/licenses/LICENSE-2.0
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+*/
+
+package com.amazonaws.services.neptune.util;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for gzip compression of CSV files
+ */
+public class GzipCompressUtils {
+    /**
+     * Upload Neptune vertices and edges CSV files asynchronously
+     */
+    public static void compressCsvFiles(String uploadDirectoryPath) throws Exception {
+        System.err.println("Compressing .csv to .gzip...");
+
+        File directory = new File(uploadDirectoryPath);
+        if (!directory.exists() || !directory.isDirectory()) {
+            throw new IllegalStateException("Directory does not exist: " + uploadDirectoryPath);
+        }
+
+        File[] csvFiles = directory.listFiles((dir, name) ->
+            name.toLowerCase().endsWith(".csv"));
+
+        if (csvFiles == null || csvFiles.length == 0) {
+            System.err.println("No CSV files found in directory: " + uploadDirectoryPath);
+            return;
+        }
+
+        System.err.println("Starting concurrent compression of " + csvFiles.length + " CSV files...");
+
+        List<CompletableFuture<String>> compressionFutures = Arrays.stream(csvFiles)
+            .map(file -> CompletableFuture.supplyAsync(() -> {
+                try {
+                    return compressSingleCsvFile(file.getAbsolutePath());
+                } catch (IOException e) {
+                    System.err.println("Failed to compress " + file.getName() + ": " + e.getMessage());
+                    return null;
```

Comment:
Is there tests for this case where null is returned? It seems like it would return an empty list of compressedFiles, but convertCSV doesn't seem to do any handling in this case?